TopPlaces/TopPlaces/PhotosHistory.m

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2016 Lightricks. All rights reserved.
+// Created by Gennadi Iosad.
+
+#import "PhotosHistory.h"
+#import "FlickrFetcher.h"
+NS_ASSUME_NONNULL_BEGIN
+#define PHOTOS_HISTORY_PREF_KEY @"history"
+#define HISTORY_LOG_LENGTH 20
+@implementation PhotosHistory
+
++ (NSArray *) historyArray
+{
+  NSUserDefaults *prefs = [NSUserDefaults standardUserDefaults];
+  NSArray *photos = [prefs objectForKey:PHOTOS_HISTORY_PREF_KEY];
```

Comment:
<code>objectForKey:</code> may return <code>nil</code>, you should mark this method as nullable, or somehow validate that you never  return nil.